### PR TITLE
Add the global timeout for search queries

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

Refactor timeouts for client and backend to use values globally defined as class constants rather than hardcoded values.

Utilize `ep_post_formatted_args` filter to set the timeout on ES post search queries.

### Plugin Updated: Enterprise Search

We've implemented a timeout on Elasticsearch queries, it matches the Enterprise Search client timeout so there's not going to be a difference in terms of behavior. It's meant as a safety mechanism to prevent runaway queries after the client has already disconnected.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1. Check out PR.
2. Do a search query
1. Verify in Search Dev Tools that payload has `timeout` in it.

### CODE REVIEW

File: search/includes/classes/class-search.php [hunk 1]
```diff
@@ -1502,15 +1507,27 @@ public function filter__jetpack_widgets_to_include( $widgets ) {
...
 		// Check if track_total_hits is set
 		// Don't override it if it is
 		if ( ! array_key_exists( 'track_total_hits', $formatted_args ) ) {
 			$formatted_args['track_total_hits'] = true;
 		}
 
+		// TODO: remove temporary rollout environment check
+		// Force the timeout for post search queries.
+		$is_prod = defined( 'VIP_GO_APP_ENVIRONMENT' ) && 'production' === VIP_GO_APP_ENVIRONMENT;
+		if ( ! isset( $formatted_args['timeout'] ) && apply_filters( 'vip_search_force_global_timeout', $is_prod ) ) {
```
Comment (rebeccahum) [inline search/includes/classes/class-search.php:1527]: Are you intending to roll it out only for production? Or are you returning false somewhere with the filter?

**Review by rebeccahum — CHANGES_REQUESTED** [review]

File: search/includes/classes/class-search.php (hunk 1 above)
Comment (rinatkhaziev) [reply search/includes/classes/class-search.php:1527]: nice catch!

**Review by rebeccahum — APPROVED** [review]

**Review by rebeccahum — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
r2166-stacks

**Comment by sjinks** [comment]
v20220111.1

**Pull request merged**